File: 2019/php/src/Aoc/Days/Day5.php
<?php


namespace Ppx17\Aoc2019\Aoc\Days;


use Ppx17\Aoc2019\Aoc\Days\Day5\IntCode;

class Day5 extends AbstractDay
{
    private IntCode $intCode;

    public function dayNumber(): int
    {
        return 5;
    }

    public function setUp(): void
    {
        parent::setUp();
        $this->intCode = new IntCode(array_map('intval', explode(',', $this->getInput())));
    }

    public function part1(): string
    {
        $this->intCode->input = 1;
        $this->intCode->run();
        return (string)$this->intCode->output;
    }

    public function part2(): string
    {
        $this->intCode->reset();
        $this->intCode->input = 5;
        $this->intCode->run();
        return (string)$this->intCode->output;
    }
}

File: 2019/julia/src/day5.jl
include("Aoc.jl")
using .Aoc
include("Intcode.jl")
using .Intcode

intcode = Aoc.intcode(5)

function run(input)
    computer = Computer(copy(intcode))
    computer.input = [input]
    Intcode.run!(computer)
    computer.output
end

part1() = run(1)
part2() = run(5)

println("Part 1: ", part1())
println("Part 2: ", part2())